gui/src/granite/GraniteWizard.tsx

```diff
@@ -391,8 +419,9 @@ const ModelSelectionStep: React.FC<StepProps> = (props) => {
     setModelInstallationError,
     systemInfo,
     statusByModel,
+    isOllamaOutdated
   } = useWizardContext();
-  const [systemError, setSystemError] = useState<string | undefined>();
+  const [systemErrors, setSystemErrors] = useState<string[] | undefined>();
```

Comment:
While systemErrors is a now an array, it doesn't seem like there is any mechanism for "cooperation" for different parts of the code - they just independently set it to [] or [<error>]. Is this actually any better?

Could you do something like:

```
 systemErrors, updateSystemErrors = useReducer((errors, { key: string, message: string | undefined}) => {
     errors[key] = message;
 }, {}) 

updateSystemErrors({ 
    key: 'ollamaOutdated',
    message: ollamaIsOutdated ? `Ollama v${MIN_OLLAMA_VERSION} is required[...] } : undefined
 });
 ```

I'm not so much worried about displaying two errors at once,  but the current code seems at least potentially racy - an updated status for Ollama resulting in the out-of-disk space message being removed or something.

Reply:
In this particular case, using useReducer would not be better, as systemErrors are actually section-scoped. There's no risk of losing any message, see https://github.com/Granite-Code/continue-for-granite/pull/25/files#diff-4a38dbbfe686ca662f3adc3cf3d43931747bcb4cf01c0379909a3a1a7f03dbe1R466-R483

Reply:
Ah, I see - I got confused by the two chunks of code setting the ollama error message. That duplication isn't great, but I don't have a good suggestion for doing it better within the current structure, so good enough for now.